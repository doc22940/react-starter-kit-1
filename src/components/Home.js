import React, { Component } from 'react'
import { CenteredDiv } from '../styles/Common'

export default class Home extends Component {
  render() {
    return (
      <CenteredDiv style={{padding: 50}}>
        To get started, edit src/components/Home.js and save to reload.
      </CenteredDiv>
    )
  }
}
