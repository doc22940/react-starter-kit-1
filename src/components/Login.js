import React, { Component } from 'react'
import {connect} from 'react-redux'
import { CenteredDiv } from '../styles/Common'
import { LoginPanel } from '../styles/Login'
import TextField from 'material-ui/TextField'
import RaisedButton from 'material-ui/RaisedButton'

import {loginRequest} from '../actions/login_actions'

class Login extends Component {
  state = { email: '', password: '' }

  updateField = (paramName, event) => this.setState({ [paramName]: event.target.value })

  submit = () => {
    this.props.loginRequest(this.state)
  }

  componentWillReceiveProps = (nextProps) => {
    console.log(nextProps.loginResponse)
  }

  render() {
    const { email, password } = this.state
    return (
      <CenteredDiv>
        <LoginPanel rounded={false}>
          <TextField value={email} floatingLabelText="Email"  onChange={this.updateField.bind(this, 'email')}/>
          <TextField value={password} floatingLabelText="Password" type="password"  onChange={this.updateField.bind(this, 'password')}/>
          <RaisedButton label="SUBMIT" backgroundColor="#a4c639" primary style={{ marginTop: 10 }} onClick={this.submit}/>
        </LoginPanel>
      </CenteredDiv>
    )
  }
}

const mapStateToProps = state => ({loginResponse: state.loginReducer})

export default connect(mapStateToProps, { loginRequest })(Login)
