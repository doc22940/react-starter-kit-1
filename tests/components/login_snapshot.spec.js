import React from 'react'
import renderer from 'react-test-renderer'
import { Login } from '../../src/components/Login'

describe('<Login/>', () => {
  it('snapshot test', () => {
    const component = renderer.create(<Login login={{ requesting: false, signed_in: false, error: null }} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('snapshot progress bar behavior', () => {
    const component = renderer.create(<Login login={{ requesting: true, signed_in: false, error: null }} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
