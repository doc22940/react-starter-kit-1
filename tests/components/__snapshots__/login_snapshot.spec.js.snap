exports[`<Login/> snapshot progress bar behavior 1`] = `
<div
  className="row">
  <div
    className="col-sm" />
  <div
    className="col-sm-12 col-md-8 col-lg-8 jxwKTr">
    <form>
      <div
        className="input-group fluid">
        <input
          className="evyRiw"
          id="emailField"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value="" />
      </div>
      <div
        className="input-group fluid">
        <input
          className="evyRiw"
          id="passwordField"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value="" />
      </div>
      <div
        className="input-group fluid dRWiaS" />
      <div
        className="input-group fluid">
        <div
          className="mLiZo">
          <div
            className="spinner-donut" />
        </div>
      </div>
    </form>
  </div>
  <div
    className="col-sm" />
</div>
`;

exports[`<Login/> snapshot test 1`] = `
<div
  className="row">
  <div
    className="col-sm" />
  <div
    className="col-sm-12 col-md-8 col-lg-8 jxwKTr">
    <form>
      <div
        className="input-group fluid">
        <input
          className="evyRiw"
          id="emailField"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value="" />
      </div>
      <div
        className="input-group fluid">
        <input
          className="evyRiw"
          id="passwordField"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value="" />
      </div>
      <div
        className="input-group fluid dRWiaS" />
      <div
        className="input-group fluid">
        <input
          className="primary cklcwZ"
          id="submitButton"
          onClick={[Function]}
          type="submit"
          value="Submit" />
      </div>
    </form>
  </div>
  <div
    className="col-sm" />
</div>
`;
